designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,213 @@
+| Authors       | Creation Date | Status      | Extra |
+|---------------|---------------|-------------|---|
+| @name | date | Implementeble | - |
+
+# Auto-generate Helm Chart Plugin
+===================
+
+The proposal is to add a new alpha command to auto-generate a helm chart for a given kustomize directory created by kubebuilder.
+
+## Example
+
+By running the following command, the plugin will generate a helm chart from the specific kustomize directory and output it to the directory specified by the `--output` flag.
+
+```shell
+kubebuilder alpha generate-helm-chart --from=<path> --output=<path>
+```
+
+## Open Questions [optional]
+
+TBF
+
+## Summary
+
+TBF
+
+## Motivation
+
+Currently, kubebuilder only supports the distribution of the operator by the kustomize. However, many well-known operators contains the distribution of helm charts:
+
+- [mongodb](https://artifacthub.io/packages/helm/mongodb-helm-charts/community-operator
+)
+- [cert-manager](https://cert-manager.io/v1.6-docs/installation/helm/#1-add-the-jetstack-helm-repository)
+- [prometheus](https://bitnami.com/stack/prometheus-operator/helm)
+
+The proposed command will auto-generate a helm chart from a kustomize directory created by kubebuilder with minimizing the manual effort. The generated helm chart will cover the frequently-used configurations.
+
+Context:
+
+- See the discussion [Is there a good solution/tool to generate a new helm chart from the manifests?](https://github.com/kubernetes-sigs/kubebuilder/discussions/3074)
+
+### Goals
+
+- Help developers to generate a helm chart from a kustomize directory.
+- Generate a helm chart without human interactions once the kustomize directory updated.
+- Reserve the ability to customize the helm chart by the developer.
+
+### Non-Goals
+
+- Generate the helm chart from a kustomize directory not created by kubebuilder.
+- Generate the fine-grained configurations.
+
+## Proposal
+
```

Comment:
```suggestion
The new optional plugin will create a new directory, deploy/helmchart, where the scaffolding of the project should be constructed as follows:

//TODO: We need a better definition 
// We must create a helmChart with the default Kubebuilder scaffold to add here

```shell
❯ tree deploy/helmchart
helmchart
    ├── Chart.yaml
    ├── crds
    │   ├── <CRDs YAML files generate under config/crds/>
    ├── templates
    │   ├── deployment.yaml 
    │   ├── _helpers.tpl
    │   ├── role_binding.yaml
    │   ├── role.yaml 
    │   ├── <all under config/rbac less the edit and view roles which are helpers for admins>
    └── values.yaml
```

```

We need to create a HelmChart based on the default scaffold provide by Kubebuilder
We can use for example the helmfy to see if that can help as out.
So that we can properly update and add the definitions here.